core/src/main/scala/kafka/server/BrokerToControllerChannelManagerImpl.scala

```diff
@@ -129,6 +137,27 @@ class BrokerToControllerChannelManagerImpl(metadataCache: kafka.server.MetadataC
     requestQueue.put(BrokerToControllerQueueItem(request, callback))
     requestThread.wakeup()
   }
+
+  override def forwardRequest(responseToOriginalClient: (RequestChannel.Request, Int =>
+                                AbstractResponse, Option[Send => Unit]) => Unit,
+                              request: RequestChannel.Request,
+                              callback: Option[Send => Unit] = Option.empty): Unit = {
+    val serializedRequestData = request.body[AbstractRequest].serialize(request.header)
```

Comment:
We want to avoid this serialization since it introduces the possibility for the request to be altered by the forwarding broker. The `RequestChannel.Request` object retains the reference to the original buffer, which we can use here, but we need to tell the channel to delay releasing the buffer using `ApiKeys.requiresDelayedAllocation` for all of the "forwardable" APIs.